### PR TITLE
refactor: Use `async/await` in `analytics`

### PR DESCRIPTION
**Pull request by pgrzesik**
Finishes 1. of `Local` 

Addresses: #9257

### CODE REVIEW

File: lib/utils/analytics/index.js [hunk 1]
```diff
@@ -29,147 +28,148 @@ const logError = (type, error) => {
...
+          return null;
+        })
+      )
     );
-  }).then(() => null); // Do not leak any result
+  } catch (readdirError) {
+    if (readdirError.code !== 'ENOENT') logError('Cannot access cache dir', readdirError);
+  }
+
+  return null;
```
Comment (medikoo) [inline lib/utils/analytics/index.js:172]: This `return` originally was intended to supress array result from `Promise.all` and here it's moved, and no longer applied in same logic point.

**Review by medikoo — CHANGES_REQUESTED** [review]
@pgrzesik looks very good! I've just noticed that `try/catch` spans are in most cases wider than in non async/await logic, which may result with some important issues being hidden from us in a future.

I've commented on those bits

File: lib/utils/analytics/index.js [hunk 2]
```diff
@@ -29,147 +28,160 @@ const logError = (type, error) => {
...
-      return null;
-    }
-  );
+const processResponseBody = async (response, id, startTime) => {
+  let result;
+
+  try {
+    result = await response.json();
+  } catch (error) {
+    --ongoingRequestsCount;
```
Comment (medikoo) [inline lib/utils/analytics/index.js:37]: To not have it repeated, this could be in `finally`

File: lib/utils/analytics/index.js [hunk 3]
```diff
@@ -29,147 +28,160 @@ const logError = (type, error) => {
...
+  }
+  if (areAnalyticsDisabled && !isForced) return null;
+  if (!cacheDirPath) return null;
+  const limit = pLimit(3);
+  let dirFilenames;
+  try {
+    dirFilenames = await fse.readdir(cacheDirPath);
+  } catch (readdirError) {
+    if (readdirError.code !== 'ENOENT') logError('Cannot access cache dir', readdirError);
+    return null;
```
Comment (medikoo) [inline lib/utils/analytics/index.js:137]: We may just `return` (no need to `return null`, same below. In old version it was `null`, to visually better signal intent (`then(() => {}` or `then(() => { return undefined }` is either confusing or dirty)

**Review by medikoo — APPROVED** [review]
Looks great :+1

I've suggested two more style improvements, but they're just style nits

**Review by medikoo — DISMISSED** [review]
Looks great 👍

**Comment by pgrzesik** [comment]
Pushed again only due to snyk check hanging - no changes were introduced from last review

**Pull request merged**